Wingstop_Recovery_Computer_V1.ino

```diff
@@ -0,0 +1,525 @@
+
+//***********LED FUNCTIONS***********
+
+void turn_on_led(){ // Turns on LED
+  digitalWrite(LED_PIN, HIGH);
+}
+
+void turn_off_led(){ // Turns off LED
+  digitalWrite(LED_PIN, LOW);
+}
+
+
+//***********BUZZER FUNCTIONS***********
+
+void turn_on_buzzer(){ // Turns on LED
+  digitalWrite(BUZZER_PIN, HIGH);
+}
+
+void turn_off_buzzer(){ // Turns off LED
+  digitalWrite(BUZZER_PIN, LOW);
+}
+
+void beep_buzz(int num){ // Blinks LED and Beeps Buzzer for num times
+  for(int i = 0; i < num; i++){
+    turn_on_buzzer();
+    turn_on_led();
+    delay(300);
+    turn_off_buzzer();
+    turn_off_led();
+    delay(300);
+  }
+}
+
+void recovery_beeps(){//Continuous Slow long beeps for recovery
+  while(1){ //does not exit loop
+  turn_on_buzzer();
+  delay(1000); //beep delay
+  turn_off_buzzer();
+  delay(5000); //pause delay
+  }
+}
+
+
+//***********BAROMETER FUNCTIONS***********
+
+void read_barometer(){//takes barometer reading
+  while (! bmp.performReading()) {//performs barometer reading
+    //could not perform reading
+  }
+}
+
+float read_altitude(){//reads and returns barometer altitude reading
+  while (! bmp.performReading()) {
+    //performs until it gets a valid reading
+  }
+  return bmp.readAltitude(init_pressure); // returns current altitude reading
+}
+
+void iterate_altitude(){ //reads altitude at frequency (hz)
+  do{
+  t2 = millis();
+  }while(t2 - t1 < 1000/hz);//runs until reaches frequency
+
+  landing = x1; //updates landing variable for detection
+  x1 = read_altitude(); //updates altitude
+  t1 = t2;  //updates new time
+}
+
+
+//***********FILE FUNCTIONS***********
+
+void open_file(){// opens the file for writing
+  myFile = SD.open("flight.txt", FILE_WRITE);
+
+  
+  if (myFile) {
+    //file opened ok
+  } else {
+    // if the file didn't open, turn on buzzer/LED to indicate file problem
+    turn_on_led(); //Turn on LED
+    turn_on_buzzer(); //Turn on buzzer
+    while(1){ 
+      //run while loop forever
+    }
+  }
+}
+
+void set_header_file(){//sets headers at begining of file
+  myFile.print(F("Flight Log:\t"));
+  myFile.print(hz);
+  myFile.println(F("hz"));
+  myFile.print(F("Time:\tAlt:\tApo:\tEvents:\t"));
+}
+
+void close_file(){
+  myFile.close(); //closes file
+}
+
+void reopen_file(){//closes then reopens the file, saving data up to this point
+  close_file();//closes the file
+  open_file();//opens the file
+}
+
+void log_data(){//saves current data to sd card
+  myFile.print("\n");
+  myFile.print(t2); myFile.print(F("\t"));       //logs current time
+  myFile.print(x1); myFile.print(F("\t"));       //logs current position
+  
+}
+
+
+
+
+//***********PHYSICS FUNCTIONS***********
+
+
+
+int detect_take_off(){//returns 1 if take off is detected, otherwise 0
+  if(x1 >= TAKEOFF_ALTITUDE){
+    return 1;
+  }
+  else{
+    return 0;
+  }
+}
+
+int detect_apogee(){//looks for apogee and returns 1 if detected, 0 if not.
```

Comment:
Instead of describing what each line does (which is redundant since each line is pretty straightforward by itself), describe the overall algorithm in a comment above the function.